garage/tf/algos/ddpg.py

```diff
@@ -0,0 +1,372 @@
+        f_init_target()
+
+        observation = self.env.reset()
+        if self.es:
+            self.es.reset()
+
+        episode_reward = 0.
+        episode_step = 0
+        episode_rewards = []
+        episode_steps = []
+        episode_actor_losses = []
+        episode_critic_losses = []
+        episodes = 0
+        epoch_ys = []
+        epoch_qs = []
+
+        for epoch in range(self.n_epochs):
+            logger.push_prefix('epoch #%d | ' % epoch)
+            logger.log("Training started")
+            for epoch_cycle in pyprind.prog_bar(range(self.n_epoch_cycles)):
+                for rollout in range(self.n_rollout_steps):
+                    action = self.es.get_action(rollout, observation,
+                                                self.actor)
+                    assert action.shape == self.env.action_space.shape
+
+                    next_observation, reward, terminal, info = self.env.step(
+                        action)
+                    episode_reward += reward
+                    episode_step += 1
+
+                    replay_buffer.add_transition(observation, action,
+                                                 reward * self.reward_scale,
+                                                 terminal, next_observation)
+
+                    observation = next_observation
+
+                    if terminal:
+                        episode_rewards.append(episode_reward)
+                        episode_steps.append(episode_step)
+                        episode_reward = 0.
+                        episode_step = 0
+                        episodes += 1
+
+                        observation = self.env.reset()
+                        if self.es:
+                            self.es.reset()
+
+                for train_itr in range(self.n_train_steps):
+                    if replay_buffer.size >= self.min_buffer_size:
+                        critic_loss, y, q, action_loss = self._learn()
+
+                        episode_actor_losses.append(action_loss)
+                        episode_critic_losses.append(critic_loss)
+                        epoch_ys.append(y)
+                        epoch_qs.append(q)
+
+            logger.log("Training finished")
+            if replay_buffer.size >= self.min_buffer_size:
+                logger.record_tabular('Epoch', epoch)
+                logger.record_tabular('Episodes', episodes)
+                logger.record_tabular('AverageReturn',
+                                      np.mean(episode_rewards))
+                logger.record_tabular('StdReturn', np.std(episode_rewards))
+                logger.record_tabular('AveragePolicyLoss',
+                                      np.mean(episode_actor_losses))
+                logger.record_tabular('AverageQFunctionLoss',
+                                      np.mean(episode_critic_losses))
+                logger.record_tabular('AverageQ', np.mean(epoch_qs))
```

Comment:
Please namespace your logs according to a relevant component.

For instance, some things are a property of the Q function and can be logged under a QFunction namespace (e.g. "QFunction/AverageQLoss"). Other things are a property of the algo as a whole ans can be left at the top level (e.g. AverageReturn").